### PR TITLE
Added opentera teleop demo

### PR DESCRIPTION
**Pull request by philippewarren**

### CODE REVIEW

File: ros/demos/ttop_opentera/launch/ttop_opentera.launch [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<launch>
+    <include file="$(find t_top)/launch/opentera_teleop.launch">
+        <arg name="visualization" value="true"/>
+    </include>
+
+    <node pkg="ttop_opentera" type="ttop_opentera_node" name="ttop_opentera_node" output="screen" launch-prefix="bash -c 'sleep 20; $0 $@' ">
+    </node>
```
[the remark this answers is not in the document]
Comment (philippewarren) [reply ros/demos/ttop_opentera/launch/ttop_opentera.launch]: Corrigé

File: ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp [hunk 2]
```diff
@@ -123,3 +126,37 @@ void HbbaLite::updateStrategies(vector<unique_ptr<Desire>> desires)
...
+                }
+                break;
+            }
+        }
+    }
+
+    return activeStrategies;
+}
+
+std::vector<std::string> HbbaLite::getActiveDesires() const
```
[the remark this answers is not in the document]
Comment (philippewarren) [reply ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp]: Ok ouin, c'est pour ça le "enabled"

File: ros/t_top/launch/behaviors/dance.launch [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<launch>
+
+    <!-- Dance -->
```
[the remark this answers is not in the document]
Comment (philippewarren) [reply ros/t_top/launch/behaviors/dance.launch]: Corrigé

File: ros/hbba_lite/include/hbba_lite/core/HbbaLite.h [hunk 4]
```diff
@@ -41,6 +42,9 @@ class HbbaLite : public DesireSetObserver
     std::atomic_bool m_stopped;
     std::unique_ptr<std::thread> m_thread;
 
+    mutable std::mutex m_activeDesireNamesMutex;
```
Comment (philippewarren) [inline ros/hbba_lite/include/hbba_lite/core/HbbaLite.h:45]: Will be locked in a `const` member function

File: ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp [hunk 5]
```diff
@@ -125,6 +123,13 @@ void HbbaLite::updateStrategies(vector<unique_ptr<Desire>> desires)
     {
         m_strategiesByDesireType[get<0>(s)][get<1>(s)]->enable(get<2>(s));
     }
+
+    std::lock_guard<std::mutex> lock(m_activeDesireNamesMutex);
+    m_activeDesireNames.clear();
+    for (auto& s : strategiesToEnable)
+    {
+        m_activeDesireNames.emplace(get<0>(s).name());
+    }
```
Comment (philippewarren) [inline ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp:132]: Add names to set for enabled desires

File: ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp [hunk 6]
```diff
@@ -149,14 +155,8 @@ std::vector<std::string> HbbaLite::getActiveStrategies() const
...
-    vector<string> activeDesires;
-
-    for (const auto& desire : m_desireSet->getEnabledDesires())
-    {
-        activeDesires.emplace_back(desire->type().name());
-    }
-
-    return activeDesires;
+    std::lock_guard<std::mutex> lock(m_activeDesireNamesMutex);
+    return {std::begin(m_activeDesireNames), std::end(m_activeDesireNames)};
```
Comment (philippewarren) [inline ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp:161]: Return as a vector with copied data, with the copy made under the locked mutex

File: ros/user_interface/CMakeLists.txt [hunk 7]
```diff
@@ -197,3 +197,17 @@ add_custom_target(
...
+function(assert_program_installed PROGRAM)
+    find_program(prog-${PROGRAM} ${PROGRAM})
+    if(${prog-${PROGRAM}} STREQUAL "prog-${PROGRAM}-NOTFOUND")
+        message(FATAL_ERROR "'${PROGRAM}' needs to be installed")
+    endif()
+endfunction()
+
+assert_program_installed("npm")
+assert_program_installed("tar")
+assert_program_installed("md5sum")
```
Comment (philippewarren) [inline ros/user_interface/CMakeLists.txt:213]: Check dependencies

File: ros/t_top/launch/opentera_teleop.launch [hunk 8]
```diff
@@ -20,8 +20,8 @@
 
     <!-- Perceptions -->
     <include file="$(find t_top)/launch/perceptions/odas.launch">
-        <arg if="$(arg use_echo_cancellation)" name="odas_configuration_path" value="$(find t_top)/config/configuration_16SoundsUSB_pa_echo_cancelled.cfg"/>
-        <arg unless="$(arg use_echo_cancellation)" name="odas_configuration_path" value="$(find t_top)/config/configuration_16SoundsUSB_pa.cfg"/>
+        <arg if="$(arg use_echo_cancellation)" name="odas_configuration_path" value="$(find t_top)/config/configuration_16SoundsUSB_pulseaudio_ec.cfg"/>
+        <arg unless="$(arg use_echo_cancellation)" name="odas_configuration_path" value="$(find t_top)/config/configuration_16SoundsUSB_pulseaudio.cfg"/>
```
Comment (philippewarren) [inline ros/t_top/launch/opentera_teleop.launch:24]: Renamed files

File: ros/t_top/config/configuration_16SoundsUSB_pulseaudio.cfg [hunk 9]
```diff
@@ -14,7 +14,7 @@ raw:
     interface: {    #"arecord -l" OR "aplay --list-devices" to see the devices
         type = "pulseaudio";
         source = "alsa_input.usb-IntRoLab_16SoundsUSB_Audio_2.0-00.multichannel-input";
-        channelnames = "front-left,front-right,rear-left,rear-right,front-center,lfe,side-left,side-right,aux0,aux1,aux2,aux3,aux4,aux5,aux6,aux7";
+        channelmap = ("front-left", "front-right", "rear-left", "rear-right", "front-center", "lfe", "side-left", "side-right", "aux0", "aux1", "aux2", "aux3", "aux4", "aux5", "aux6", "aux7");
```
Comment (philippewarren) [inline ros/t_top/config/configuration_16SoundsUSB_pulseaudio.cfg:17]: Removed support for the comma separated list as one string in odas

**Review by mamaheux — APPROVED** [review]

File: ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp [hunk 10]
```diff
@@ -122,4 +123,40 @@ void HbbaLite::updateStrategies(vector<unique_ptr<Desire>> desires)
...
+
+    std::lock_guard<std::mutex> lock(m_activeDesireNamesMutex);
+    m_activeDesireNames.clear();
+    for (auto& s : strategiesToEnable)
+    {
+        m_activeDesireNames.emplace(get<0>(s).name());
+    }
+}
+
+std::vector<std::string> HbbaLite::getActiveStrategies() const
```
Comment (mamaheux) [inline ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp:135]: Ajouter un test

File: ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp [hunk 11]
```diff
@@ -122,4 +123,40 @@ void HbbaLite::updateStrategies(vector<unique_ptr<Desire>> desires)
...
+                }
+                break;
+            }
+        }
+    }
+
+    return activeStrategies;
+}
+
+std::vector<std::string> HbbaLite::getActiveDesireNames() const
```
Comment (mamaheux) [inline ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp:158]: Ajouter un test

File: ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp [hunk 12]
```diff
@@ -122,4 +123,40 @@ void HbbaLite::updateStrategies(vector<unique_ptr<Desire>> desires)
...
+{
+    std::lock_guard<std::mutex> lock(m_activeDesireNamesMutex);
+    return {std::begin(m_activeDesireNames), std::end(m_activeDesireNames)};
```
[the remark this answers is not in the document]
Comment (mamaheux) [reply ros/hbba_lite/src/hbba_lite_cpp/core/HbbaLite.cpp]: I didn't know that m_activeDesireNames is a set.
You can return a vector.

File: ros/behaviors/teleoperation/scripts/teleoperation_node.py [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+                    0,
+                    0,
+                    # TODO: replace 'speed' by 'self._state.current_speed.head_angle' when the neural network is replaced by direct kinematics
+                    speed,
+                    0
+                ],
+                should_sleep=False,
+                current_head_pose=[0, 0, HEAD_ZERO_Z, 0,
+                                   self._movement_commands.current_head_pose[4], 0]
+            )
```
[the remark this answers is not in the document]
Comment (philippewarren) [reply ros/behaviors/teleoperation/scripts/teleoperation_node.py]: Bon point, ce n'est plus nécessaire. J'ai enlevé les TODO

**Review by mamaheux — APPROVED** [review]

**Pull request merged**